flink/src/main/java/org/apache/iceberg/flink/sink/IcebergStreamWriter.java

```diff
@@ -62,7 +62,7 @@ public void open() {
   @Override
   public void prepareSnapshotPreBarrier(long checkpointId) throws Exception {
     // close all open files and emit files to downstream committer operator
-    for (DataFile dataFile : writer.complete()) {
+    for (DataFile dataFile : writer.complete().dataFiles()) {
```

Comment:
Should this also check that there are no delete files?

Reply:
Yes,  it should be emit the `deleteFiles` to downstream, and commit both delete files and data files into iceberg table by `RowDelta` API.  I think that would be a separate PR to address this.